rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,157 @@
+   is announced to the community. Barring objections or last-minute
+   fixes judged valid by the release manager (or core team), the
+   commit is tagged and the branch goes into the maintenance phase.
+   Readiness is a judgment call, and should require increased
+   scrutiny/validation for a release with more complex/major changes.
+8. So long as no new feature work has yet happened on `master`,
+   `master` should be kept equal with the most recent `#-maintenance`
+   branch (through fast-forwards, not cherry-picks). Once new feature
+   work occurs, the semver minor (or major) versions are bumped and
+   normal development continues against `master`. This allows us to
+   maintain the "no patch release without a corresponding mainline
+   release" invariant (see point 13 below) while still allowing patch
+   releases to happen on the appropriate `#-maintenance` branch.
+9. Associated with each `NixOS` stable release is an associated
+   `maintained` series of `Nix`, with the same support lifetime as
+    the stable release.
+10. Any important bugfix can be cherry-picked from `master` to a
+    `#-maintenance` branch of a supported `maintained` series at any
+    committer's discretion.
+11. In rare cases, if a change truly only makes sense on a
+    `#-maintenance` branch (e.g. when we added support to `1.11` to
+    be able to work with a `2.0` database schema) it can be targeted
+    directly to the `#-maintenance` branch. This should be avoided if
+    possible and subject to especially careful testing and review.
+12. A maintenance release can be tagged off of the `#-maintenance`
+    branch at any time *after* there is a mainline release that
+    contains all of the fixes included in the desired maintenance
+    release, at the discretion of the Nix core team. This implies that
+    maintenance releases may be tagged off of commits behind the tip
+    of the maintenance branch, if the latest commits haven't been
+    included in a mainline release.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The main drawback to having a formalized model in general is our model
+may not match a reasonable productive process in practice and so may
+lead to obnoxious busy work or delays for no purpose.
+
+The drawbacks of this specific model are a bit hard to see without
+putting it into practice. One thing that seems apparent is there is a
+non-trivial amount of complexity (12 bullet points!) which could
+perhaps be reduced. It also may prove difficult under this model to
+develop especially large features, as without the right development
+practices (e.g. feature flags on `master`) this can tend to create
+new long-lived branches that can effectively become the new `master`.
+
+# Alternatives
+[alternatives]: #alternatives
+
+If we don't do anything at all, we are likely to have more long-lived
+unreleased branches and significant undertested divergence, as well as
+lack of guidance to the Core team for how to handle release requests.
```

Comment:
Given that the Nix Core team has to approve this policy first, I would say «lack of a clear policy» instead of «lack of guidance»